internal-docs/engineering/doc-site/open-graph-image-preview-function.md

```diff
@@ -0,0 +1,77 @@
+    S->>W: Do you have any open graph tags?
+    note right of S: https://paste.twilio.design/components/button
+    W-->>S: og:image url 
+    note right of S: <meta property="og:image" content={openGraphServiceUrl} />
+    S->>N: Can I have the image pls?
+    note right of S: http://paste.twilio.design/.netlify/functions/opengraph/components/button
+    loop 
+        N->>N: Figure out which component and build request params for opengraph page
+    end
+    note right of N: components/button
+    N->>W: Can I have this page pls?
+    note right of W: https://paste.twilio.design/opengraph/?path=components/button
+    W-->>N: Here you go
+    loop Screenshot
+        N->>N: Screenshot the page
+    end
+    N-->>S: The image you requested
+    S-->>U: Lovely preview of URL you shared
+```
+
+## Local development
+
+We use [Netlify CLI](https://docs.netlify.com/cli/get-started/) to run our websites locally. This allows us to run the website and [Netlify functions](https://docs.netlify.com/functions/overview/) in parallel, and Netlify CLI proxy's both to a single port. This makes it feel like to two things are running on the same domain.
+
+To start, run:
+
+```
+yarn start:website
+```
+
+Once both the website and netlify functions have started running, both will be available on `localhost:8888`.
+
+🚨 **It takes some time for Gastby to boot up. Be patient** 🚨
+
+Once running, the function can be reached at `localhost:8888/.netlify/functions/opengraph/` for testing.
+
+Whilst running, changes to the function code will restart the function with the new code changes applied. (Hot reloading for Node, using Nodemon) This is all handled for us using the Netlify CLI.
+
+## Technology / Stack
+
+Netlify Functions are just AWS Lambdas in disguise. They take an incoming request and you can return a response. 
+
+Inside our function we use `chrome-aws-lambda` for grabbing the Chrome Browser binary for AWS Lambdas, and puppeteer to run and control the browser. 
+
+Using Puppeteer, we instruct Chrome to visit a url on the website, which renders a dynamic page based on the parameters we give it. We then tell puppeteer to use Chrome to take a screenshot of the page, and that image is then returned as the response of the function.
+
+The page on the website renders a box 800px x 420px. The information is display for the component being requested comes from Airtable, via the Gatsby integration.
+
+[Winston](https://github.com/winstonjs/winston) does our logging, so we can observe the function running in real time in the Netlify UI.
+
+[Rollbar](https://docs.rollbar.com/docs/aws-lambda) wraps our function and helps log errors to the Rollbar service so we are alerted if it starts to break.
+
+## Node runtime 
+
+🚨 **Node v14** 🚨
+
+The Netlify function must run using Node v14 for the Chrome binary to work correctly. We are explicitly opting out of Node 16 runtime in Netlify functions by setting the `AWS_LAMBDA_JS_RUNTIME` environment variable in the Netlify UI.
```

Comment:
Question: what would it take for the Chrome binary to work with Node 16? Is this work that we will have to do later?

Reply:
There is a JIRA ticket for it now https://issues.corp.twilio.com/browse/DSYS-3479. What needs to be done needs to be figured out